TulparAudio/source/TulparConfigurator.cpp

```diff
@@ -0,0 +1,73 @@
+/*
+* Copyright (C) 2018 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+
+#include <tulpar/TulparConfigurator.hpp>
+
+#include <tulpar/InternalLoggers.hpp>
+
+#include <AL/alc.h>
+
+namespace tulpar
+{
+
+TulparConfigurator::TulparConfigurator()
+    : bufferBatch(32)
+    , sourceBatch(32)
+    , device(TulparConfigurator::Device
+        {
+            std::string()
+            , true
+        })
+{
+
+}
+
+std::vector<TulparConfigurator::Device> TulparConfigurator::GetDevices()
+{
+    std::vector<TulparConfigurator::Device> result;
+
+    std::string const defaultDevice(alcGetString(NULL, ALC_DEFAULT_ALL_DEVICES_SPECIFIER));
+    ALCchar const* devices = alcGetString(NULL, ALC_ALL_DEVICES_SPECIFIER);
+
+    if (NULL != devices && '\0' != (*devices))
+    {
+        ALCchar const* it = devices;
+
+        std::string name(it);
+        it += name.size() + 1;
+
+        result.push_back({name, (0 == defaultDevice.compare(name))});
+
+        while (*it != '\0')
+        {
+            name = it;
+            it += name.size() + 1;
+
+            result.push_back({name, (0 == defaultDevice.compare(name))});
```

Comment:
Emplace?

Reply:
Not possible since it's a POD that doesn't have constructors.

Reply:
Will create constructor